### DISCUSSION TITLE
To Implement latest Indian State boundaries

### DISCUSSION THREAD

**Discussion by harvir08** [dtcenter/METplus#1695] [File I/O] [answered]
While running MODE, I came across that in MET/METplus there are very old Indian State boundaries, which are outdated and should be updated. I have updated country_data at my end and working fine for the country as a whole. For smaller regions sometimes we need state boundaries as well, till now default admin_by_country/admin_India_data is not updated. I'm trying to update it using 
[India_Boundary_Updated-master.zip](https://github.com/dtcenter/METplus/files/9092286/India_Boundary_Updated-master.zip)
, I found some code on [Github](https://github.com/dtcenter/MET/blob/main_v9.1/met/src/tools/dev_utils/shapefiles/make_mapfiles.cc) and it is not working for me (Maybe I'm not that much aware of C programming). Now please help me to convert shapefiles into the map data format that MET expects.

**Comment by georgemccabe** [comment]
Hi @harvir08, Thank you for bringing this to our attention. I spoke with another METplus developer and they suggested that we update all of the map files that are provided with the MET software. I created a [GitHub issue](https://github.com/dtcenter/MET/issues/2207) to track this work. I referenced this discussion in that issue and made a note to refer to the zip file you provided. You can follow the progress of this work in the issue.

**Reply by georgemccabe** [reply]
Hi @harvir08, the work to update the map files will be completed for the MET-11.0.0-beta3 release that is currently scheduled for 2022-09-14. You can follow the progress of the task in [dtcenter/MET#2207](https://github.com/dtcenter/MET/issues/2207). I am going to lock this discussion to prevent future posts. Please create a new discussion topic if you have any other questions.

**Reply by harvir08** [reply]
Thank you @georgemccabe for the update. I will be waiting for the MET-11.0.0-beta3 release.

**Comment by JohnHalleyGotway** [comment, marked as answer]
@harvir08, FYI, I reopened this discussion. I have been working on dtcenter/MET#2207 and trying to decide a way forward there.

However, in the meantime, I wanted to followup with you about the data shapefile data you sent. I ran it through the gis_dump_dbf and make_mapfiles development utilities on it. And I'll describe that below. Note however that I had to make a couple changes on the feature_2207_map feature branch to get this to work.

1. The gis_dump_dbf utility describes the shapefile metadata:
```
gis_dump_dbf India_State_Boundary_Updated.dbf
```
The subrecord of interest is named "stname":
```
SubRecords ... 
|    SubRecord 0
|    |    field_name         = "stname"
```
2. Run make_mapefiles to convert the shapefile to a format the MET can read:
```
make_mapfiles India_State_Boundary_Updated.shp India_State_Boundary_Updated.shx India_State_Boundary_Updated.dbf stname NA
```
And that creates a file named India_State_Boundary_Updated_data. Plotting that data via MET produces these boundaries:
<img width="129" alt="Screen Shot 2022-08-12 at 2 44 02 PM" src="https://user-images.githubusercontent.com/21087144/184442248-864775b8-ace0-496f-9517-4d3744687c52.png">

Here is the data file you're looking for:
[India_State_Boundary_Updated_data.tar.gz](https://github.com/dtcenter/METplus/files/9329827/India_State_Boundary_Updated_data.tar.gz)

In my mind the ideal situation would be enhancing MET to plot maps as shapefiles instead of this custom map data format. That's what is briefly described by dtcenter/MET#1012. That would give users like you to define the map data as they see fit without relying on a custom format.

Do you agree that'd be the best solution?

**Reply by harvir08** [reply]
Thank you, Dr @JohnHalleyGotway. I agree with the solution you provided to me and the  MET community. For the time being my problem was solved. Yes, my primary idea was MET to plot maps as shapefiles instead of this custom map data format. Maybe in the future releases of METplus, it can be implemented.
